### ISSUE TITLE
Do not return unknown identifiers from the cache

### ISSUE THREAD

**Issue by emanuil-tolev** [CottageLabs/idfind#16]
If we try to identify an identifier and we fail, it gets indexed into ES as "type":"unknown". That's how it should be, because we could run a nightly script[1] on those which would then run any newly submitted tests[2]. However, when we try to look in the cache for an identifier in order to return that cache hit to the user, we need to NOT look at unknowns.

Otherwise we end up with a situation where an identifier is regarded as "unknown" only once, when it first enters the index - and from then on idfind will return "successful" for identification attempts, but all they would say is that the identifier is "type":"unknown". So, successfully identified, but we know nothing about it.

That'd be best achieved by still writing all unknowns to the index, but excluding the unknowns when we query the cache. EDIT: even simpler - write them to a different document type. Identifiers have an ES type of 'identifier', unknowns have a type of 'uidentifier'.

[1] a nightly script OR something similar, we've not actually decided on anything
[2] Note: we've got to record a timestamp on when an attempt was last made to identify the identifier. Otherwise we'd have to run all tests on all unknown identifiers, which might not be too bad in the beginning. However, it's not a good idea to have an external process that doesn't scale very well.

**Comment by emanuil-tolev** [comment]
unknowns not saved correctly when cache and regex identification fail, Emanuil to have a look when there's time

edit: unknowns (uidentifier type documents) not saved at all, despite all the correct values being in place, e.g. db name, class type ... printing the values from the upsert method reveals nothing, yet there _must_ be some issue with saving unknown identifiers, since they don't appear in the index!

Issue #4 also shows ES connection errors where the document type is "uidentifier", may not be related.

**Comment by emanuil-tolev** [comment]
Issue #4 definitely related and problem fixed (source of bug and solution described there).

We still need to decide how to handle unknown identifiers since now we've got (possibly) multiple copies of the same unknown identifier in the index :/.

I consider this issue closed though, since IDFind does not (correctly) return unknown identifiers from the cache. I'll open a new issue wrt how we handle the unknowns (#26).
